src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10108,34 +10110,69 @@ private StatementSyntax ParseLocalDeclarationStatement(SyntaxList<AttributeListS
             }
         }
 
-        private SyntaxToken ParsePossibleScopedKeyword(bool isFunctionPointerParameter)
+        private SyntaxToken ParsePossibleScopedKeyword(
+            bool isFunctionPointerParameter,
+            bool isLambdaParameter)
         {
-            if (this.CurrentToken.ContextualKind == SyntaxKind.ScopedKeyword)
-            {
-                using var beforeScopedResetPoint = this.GetDisposableResetPoint(resetOnDispose: false);
+            if (this.CurrentToken.ContextualKind != SyntaxKind.ScopedKeyword)
+                return null;
 
-                SyntaxToken scopedKeyword = this.EatContextualToken(SyntaxKind.ScopedKeyword);
+            using var beforeScopedResetPoint = this.GetDisposableResetPoint(resetOnDispose: false);
 
-                if (this.CurrentToken.Kind is not (SyntaxKind.RefKeyword or SyntaxKind.OutKeyword or SyntaxKind.InKeyword))
-                {
-                    using var afterScopedResetPoint = this.GetDisposableResetPoint(resetOnDispose: false);
+            var scopedKeyword = this.EatContextualToken(SyntaxKind.ScopedKeyword);
 
-                    if (ScanType() == ScanTypeFlags.NotType ||
-                        (isFunctionPointerParameter
-                            ? this.CurrentToken.Kind is not (SyntaxKind.CommaToken or SyntaxKind.GreaterThanToken)
-                            : this.CurrentToken.Kind != SyntaxKind.IdentifierToken))
-                    {
-                        beforeScopedResetPoint.Reset();
-                        return null;
-                    }
+            // trivial case.  scoped ref/out/in  is definitely the scoped keyword.
+            if (this.CurrentToken.Kind is (SyntaxKind.RefKeyword or SyntaxKind.OutKeyword or SyntaxKind.InKeyword))
+                return scopedKeyword;
 
-                    afterScopedResetPoint.Reset();
-                }
+            // (scoped a) =>    (scoped a, ...) =>   is legal in a lambda.
 
+            // This is an unusual use of LangVersion. In this case we are effectively making a language breaking change
+            // to consider "scoped" a modifier in all ambiguous cases. To avoid breaking older code that is not
+            // using this version we conditionally parse based on langversion
+            if (isLambdaParameter &&
+                IsTrueIdentifier(this.CurrentToken) &&
+                this.PeekToken(1).Kind is SyntaxKind.CommaToken or SyntaxKind.CloseParenToken &&
```

Comment:
>or SyntaxKind.CloseParenToken [](http://example.com/codeflow?start=64&length=29)

`... or SyntaxKind.EqualsToken`?

Reply:
not in that case (i'll doc).  `scoped s = ...` is not legal as a modifier.  The spec explicitly disallows having an equals-value with a lambda paramter if it odesn't have a type.  So in this case `scoped` is still treated as the type.  THere are tests for this case.